### ISSUE TITLE
Automl needs more user-friendly recipe

### ISSUE THREAD

**Issue by yushan111** [intel-analytics/analytics-zoo#3623]
such as a dictionary which contains search space.

**Comment by TheaperDeng** [comment]
### Recipe refactor plan

- Automl recipe will be changed to a dictionary rather than a class inherite from base recipe.
  - Hyperparameters for network architecture and training
    - dropout, num_channels, kernel_size... (ad-hoc to each build-in model/customized model)
    - lr, batch_size... (fixed name for build-in model/customized model)
    - model (fixed name for build-in model only)
  - Hyperparameters for dataloader/feature transformer
    - past_seq_len, selected_features
  - Hpterparameters for runtime
    - training_iteration, num_samples, reward_metric

- Use `zouwu.hp` (?) as an external intercface to replace `ray.tune`. 
  - https://docs.ray.io/en/master/tune/api_docs/search_space.html
- RayTuneSearcher can take recipe dictionary as parameter and construct runtime_params and search_space.

Here is an example

```python
import zouwu.hp as hp
GRID_RANDOM_RECIPE = {# -------- model selection
    		      "model": hp.choice(["LSTM", "Seq2seq"]),
                      
                      # --------- Vanilla LSTM model parameters
                      "lstm_1_units": hp.grid_search([16, 32]),
                      "dropout_1": 0.2,
                      "lstm_2_units": hp.grid_search([16, 32]),
                      "dropout_2": hp.uniform(0.2, 0.5),

                      # ----------- Seq2Seq model parameters
                      "latent_dim": hp.grid_search([32, 64]),
                      "dropout": hp.uniform(0.2, 0.5)

		      # ----------- optimization parameters
                      "lr": hp.uniform(0.001, 0.01),
            	      "batch_size": hp.choice([32, 64]),
            	      "epochs": 10,
						
		      # ----------- data related
		      "past_seq_len": hp.randint(24,84),
		      "selected_features": hp.randfeature(min=3, max=hp.ALLFEATURE)}
recipe = hp.Recipe(space=GRID_RANDOM_RECIPE, training_iteration=20, num_samples=2)
# Then use this recipe as before
```
